internal/checks/checks.go

```
@@ -0,0 +1,177 @@
+		out := outRows[i]
+
+		if !slices.Contains(portsToCheck, in.Port) {
+			continue
+		}
+
+		portCheck := result.PartialResult{Output: fmt.Sprintf("Port %v", in.Port)}
+
+		inLoss := utils.LossPercent(in.InDropPkts, in.InTotalPkts)
+		outLoss := utils.LossPercent(out.OutDropPkts, out.OutTotalPkts)
+
+		inStatus := utils.StatusByThreshold(inLoss, warn, crit)
+		outStatus := utils.StatusByThreshold(outLoss, warn, crit)
+
+		portStatus := max(inStatus, outStatus)
+		worst = max(worst, portStatus)
+
+		addPerfSubcheck := func(label string, loss float64, status int) {
+			sub := result.PartialResult{
+				Output: fmt.Sprintf("%s: %.2f%% loss", label, loss),
+			}
+			_ = sub.SetState(status)
+			sub.Perfdata.Add(&perfdata.Perfdata{
+				Label: fmt.Sprintf("port %v %s loss", in.Port, label),
+				Value: loss, Min: 0, Max: 100,
+			})
+			portCheck.AddSubcheck(sub)
+		}
+
+		addPerfSubcheck("IN", inLoss, inStatus)
+		addPerfSubcheck("OUT", outLoss, outStatus)
+
+		if err := portCheck.SetState(portStatus); err != nil {
+			return nil, err
+		}
+		overall.AddSubcheck(portCheck)
+	}
+
+	if err := overall.SetState(worst); err != nil {
+		return nil, err
+	}
+	return &overall, nil
+}
+
+// CheckPoe creates a partialResult with information about every port's POE status
+func CheckPoe(ports []netgear.PoePort) (*result.PartialResult, error) {
+	partial := result.PartialResult{Output: "Power over Ethernet Statistics"}
+	worst := check.OK
+
+	for _, port := range ports {
+		state := "disabled"
+		if port.Enable {
+			state = "enabled"
+		}
+
+		status := check.OK
+		if port.CurrentPower > port.PowerLimit {
+			status = check.Critical
+		} else if port.CurrentPower == port.PowerLimit {
+			status = check.Warning
+		}
+
+		worst = max(worst, status)
+
+		poeCheck := result.PartialResult{
+			Output: fmt.Sprintf(
+				"Port %v is %v. Current power: %.2f/%.2fV",
+				port.Port, state, port.CurrentPower/1000, port.PowerLimit/1000,
+			),
+		}
+		_ = poeCheck.SetState(status)
```

Still ignoring an error.